### DISCUSSION TITLE
mitmproxy don't work with bore tunnel and own DNS server

### DISCUSSION THREAD

**Discussion by trueblablak** [mitmproxy/mitmproxy#6706] [General]
#### Problem Description
Cannot connect to google.com via HTTPS from mobile phone with this setup:

#### Steps to reproduce the behavior:
1. Install mitmproxy's CA as trusted by system on an Android phone
2. Run mitmproxy on some machine
3. Run DNS server that will return IP of mitmproxy for every query (in my case https://github.com/SySS-Research/dns-mitm, on server 1):
```
python dns-mitm.py '.*,<IP of mitmproxy>'
```
5. Configure phone to use that DNS server (Settings -> WiFi -> <Your WiFI> -> Edit -> IP -> Static -> DNS = IP of your DNS)
6. Run a bore server somewhere (in my case on server 1, requires root IIRC): 
```
bore server --min-port 0
```
7.  Create a tunnel for access machine with running mitmproxy from machine with running bore server (run this on mitmproxy host machine):
```
bore local 443 -p 443 --to <IP of bore server>
```
8. Every app on phone now cannot access internet with HTTPS

Bore is used to prevent usage of NAT.
Bore github [repo](https://github.com/ekzhang/bore).

#### System Information
```
mitmproxy --version        
Mitmproxy: 10.2.2 binary
Python:    3.12.1
OpenSSL:   OpenSSL 3.1.4 24 Oct 2023
Platform:  Linux-6.1.10-gentoo-dist-x86_64-Intel-R-_Core-TM-_i5-1035G1_CPU_@_1.00GHz-with-glibc2.38
```

**Comment by mhils** [comment]
This looks like an error in your routing configuration and not a mitmproxy bug. My recommendation would be to use mitmproxy's WireGuard mode and ditch all the extra components.
